### PR TITLE
Contracts/feature/new specs tulip testing

### PR DESCRIPTION
**Pull request by W3stside**
Added some useful features and changed a couple functions + logging
1. any `number` can now be converted via e.q `(50).toWei()` = `// 50000000000000000` or `(5000000000000).toEth()` = `// 50` 
2. added 3rd, optional param in `setupTests()` to pass in amounts for ETH or GNO to deposit, approve and asserting these are correct
3. cleaned up logs to read easier
4. other things???

### CODE REVIEW

File: test/dutchExchange-Tulip.js [hunk 1]
```diff
@@ -42,14 +43,29 @@ const setupContracts = async () => {
...
-const getBalance = async (acct, token) => ((await dx.balances.call(token.address, acct)).toNumber())
+// I know, it's gross
+// add wei converter
+/* eslint no-extend-native: 0 */
+Number.prototype.toWei = function toWei() {
+  return (this * 10 ** 18)
+}
+Number.prototype.toEth = function toEth() {
+  return (this / 10 ** 18)
+}
```
Comment (Velenir) [inline test/dutchExchange-Tulip.js]: this still leaves big number errors in cases:
```
> (200000000000000001).toEth()
0.2

> (4.1).toWei()
4099999999999999500
```
As long as you stick to `40|50` and the like numbers it's all good, but maybe consider `web3.fromWei/toWei`

File: test/testFunctions.js [hunk 2]
```diff
@@ -230,13 +294,14 @@ const checkUserReceivesTulipTokens = async (ST, BT, user) => {
...
+    USER'S LOCKED TUL AMT = ${lockedTulFunds.toEth()}
 
-    USER'S ETH AMT = ${newBalance}
+    USER'S ETH AMT = ${newBalance.toEth()}
   `)
-  // due to passage of time(stamp)
-  assert.isAtLeast(lockedTulFunds, tulips, 'final tulip tokens are slightly > than calculated from dx.claimBuyerFunds.call')
+  // with changes, TULIPS are NOT minted until auctionCleared
+  // lockedTulFunds should = 0
+  assert.isAtLeast(lockedTulFunds, 0, 'final tulip tokens are slightly > than calculated from dx.claimBuyerFunds.call')
```
Comment (josojo) [inline test/testFunctions.js]: Could you please check here the exact amount?assert.isAtLeast->assert.equals.  I think it is needed to be sure everything worked

File: test/dutchExchange-Tulip.js [hunk 3]
```diff
@@ -42,14 +54,188 @@ const setupContracts = async () => {
...
+    // await postBuyOrder(eth, gno, 1, 400..toWei(), buyer1)
+    // just to close auction
+    await claimSellerFunds(eth, gno, seller1, 1)
+    // await checkUserReceivesTulipTokens(eth, gno, buyer1)
+    await unlockTulipTokens(seller1)
+  })
+
+  after(() => {
+    eventWatcher.stopWatching()
+  })
```
Comment (Velenir) [inline test/dutchExchange-Tulip.js]: Can just be `after(eventWatcher.stopWatching)`
+ on line 435

File: test/dutchExchange-Tulip.js [hunk 4]
```diff
@@ -42,14 +54,188 @@ const setupContracts = async () => {
...
+      0,              // buyVolume for GNO
+      2,              // lastClosingPrice NUM
+      1,              // lastClosingPrice DEN
+      { from: seller1 },
+    )
+    seller1Balance = await getBalance(seller1, eth)
+    assert.equal(seller1Balance, (40).toWei(), 'Seller1 should have 40 balance after new Token Pair add')
+  })
+  it('Check sellVolume', async () => {
+    console.log(`
```
Comment (Velenir) [inline test/dutchExchange-Tulip.js]: Can you replace `console.log` with `log` from `utils.js` so that `--silent` works

**Comment by Velenir** [comment]
Running tests twice back-to-back breaks on the second run with
```
Error: Error: sender doesn't have enough funds to send tx. The upfront cost is: 90670000000000000000 and the sender's account only has: 9932345200000000000
```

File: test/testFunctions.js [hunk 5]
```diff
@@ -205,56 +303,70 @@ const checkUserReceivesTulipTokens = async (ST, BT, user) => {
...
   assert.isAtLeast(newBalance, lockedTulFunds, 'for ETH -> * pair returned tokens should equal tulips minted')
 }
 
 /**
  * unlockTulipTokens
  * @param {address} user => address to unlock Tokens for
  */
-const unlockTulipTokens = async (user) => {
+const unlockTulipTokens = async (user, amt = undefined) => {
   const { TokenTUL: tokenTUL } = await getContracts()
```
Comment (Velenir) [inline test/testFunctions.js]: `amt = undefined` doesn't do a thing. `amt` will be `undefined` if it's not passed anyway.
Also it's never used in the function body

File: test/testFunctions.js [hunk 6]
```diff
@@ -28,34 +49,68 @@ const getContracts = async () => {
...
-  await oracle.updateETHUSDPrice(60000, { from: accounts[0] })
+  await oracle.updateETHUSDPrice(ethUSDPrice, { from: accounts[0] })
+
+  const gnoAcctBalances = await Promise.all(accounts.map(accts => getBalance(accts, gno)))
+  const ethAcctBalances = await Promise.all(accounts.map(accts => getBalance(accts, eth)))
+
+  gnoAcctBalances.forEach((bal, i) => {
+    if (i === 0) return
+    assert.equal(bal, startingGNO)
+  })
```
Comment (Velenir) [inline test/testFunctions.js]: An alternative code would be:
```
gnoAcctBalances.slice(1).forEach(bal => assert.equal(bal, startingGNO))
```
but whichever you prefer.

File: test/testFunctions.js (hunk 5 above)
Comment (W3stside) [reply test/testFunctions.js]: @Velenir done

File: test/dutchExchange-Tulip.js (hunk 4 above)
Comment (W3stside) [reply test/dutchExchange-Tulip.js]: @Velenir done

File: test/testFunctions.js (hunk 6 above)
Comment (W3stside) [reply test/testFunctions.js]: @Velenir way cleaner. Done

**Comment by W3stside** [comment]
@josojo @Velenir

**Pull request merged**